arch/risc-v/src/esp32c3/esp32c3_wifi_adapter.h

```diff
@@ -0,0 +1,242 @@
+/****************************************************************************
+ * arch/risc-v/src/esp32c3/esp32c3_wifi_adapter.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_RISCV_SRC_ESP32C3_ESP32C3_WIFI_ADAPTER_H
+#define __ARCH_RISCV_SRC_ESP32C3_ESP32C3_WIFI_ADAPTER_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+
+#ifndef __ASSEMBLY__
+
+#undef EXTERN
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* WiFi event ID */
+
+enum wifi_adpt_evt_e
+{
+  WIFI_ADPT_EVT_STA_START = 0,
+  WIFI_ADPT_EVT_STA_CONNECT,
+  WIFI_ADPT_EVT_STA_DISCONNECT,
+  WIFI_ADPT_EVT_STA_AUTHMODE_CHANGE,
+  WIFI_ADPT_EVT_STA_STOP,
+  WIFI_ADPT_EVT_MAX,
+};
+
+/* WiFi event callback function */
+
+typedef void (*wifi_evt_cb_t)(void *p);
+
+typedef void (* wifi_tx_done_cb_t)(uint8_t ifidx, uint8_t *data,
+                                   uint16_t *len, bool txstatus);
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp_wifi_adapter_init
+ *
+ * Description:
+ *   Initialize ESP32 WiFi adapter
```

Comment:
In documentation, we should write Wi-Fi, which is the correct spelling.
Could you please fix every other occurrence?

Reply:
Of course.